---
title: "RWC23_ELT2_ChIP_Boeck_Time_Resolved_RNA"
output:
  pdf_document: default
  html_document:
    df_print: paged
  word_document:
    keep_md: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Note: Ensure BioConductor is version 3.10 or above

Install libraries

```{r}
# fill this in
# install.packages("fpc")
# install.packages("vcd")
```

Note: you must load `biomaRt` before loading `tidyverse`

Load libraries
```{r}
library(biomaRt)
library(tidyverse)
library(ComplexHeatmap)

```

Load custom functions

```{r}
source("../RWC23_Functions.R")
```


Pseudocode:
- Bring in Boeck Data
- Translate to WBGeneID
- Filter for ELT-2 ChIP bound genes, make heatmap
- Filter for intestine expressed genes (spencer data), make heatmap, add row annotation for binding cluster

# Import Time-resolved RNA

```{r data, cache=TRUE}
time_resolved_rna <-
  read.delim(
    "../02_Public_Intesine_RNA/01_input/9_Boeck_et_al_2016_time-resolved_transcriptome/Unified_dcpm_per_wormbase_gene.txt",
    quote = "",
    stringsAsFactors = FALSE
  )

paramart <-
  useMart("parasite_mart",
          dataset = "wbps_gene",
          host = "https://parasite.wormbase.org",
          port = 443)

time_resolved_rna <- getBM(
  mart = paramart,
  filter = c("wormbase_gseqname"),
  value = time_resolved_rna$WormbaseName,
  attributes = c("wormbase_gseq", "wbps_gene_id", "wikigene_name")
) %>% right_join(time_resolved_rna, by = c("wormbase_gseq" = "WormbaseName"))


time_resolved_rna <- time_resolved_rna %>% drop_na(wbps_gene_id)
```

```{r}
intestine_gene_list <-
  read_csv("../02_Public_Intesine_RNA/02_output/RWC23_Public_Intestine_RNA_Data.csv")
```


# Import wTF3.0 worm transcription factor database

```{r}
wTF3.0 <-
  read.csv(
    "../01_ChIPseq_RNAseq_Integration/01_input/TF3-0_namesonly.txt",
    sep = "\t",
    header = TRUE
  ) %>% select(WBGeneID)
```


# Filter time-resolved RNA-seq based on intestine expression

```{r}
time_resolved_rna_intestine_df <- time_resolved_rna %>%
  remove_rownames() %>%
  arrange(wbps_gene_id) %>%
  filter(wbps_gene_id %in% intestine_gene_list$WBGeneID) %>%
  select(-(emb_4cell:emb_471min),-DE,-D,-DX,-Soma,-Male,-AdultSPE9,-gonad,-LENGTH)
head(time_resolved_rna_intestine_df)
```


```{r}
time_resolved_rna_intestine_matrix <-
  time_resolved_rna_intestine_df %>%
  select(-wormbase_gseq, -wikigene_name) %>%
  remove_rownames() %>%
  arrange(wbps_gene_id) %>%
  column_to_rownames(var = "wbps_gene_id") %>%
  as.matrix()
head(time_resolved_rna_intestine_matrix)
```

Perform row normalization

```{r}
time_resolved_rna_intestine_matrix_scaled <-
  t(apply(unlist(time_resolved_rna_intestine_matrix), 1, scale))
colnames(time_resolved_rna_intestine_matrix_scaled) <-
  colnames(time_resolved_rna_intestine_matrix)
```

Store index of relavent genes for row annotations. Use custom function 

```{r}
gene_names <-
  c("elt-2", "elt-7", "elt-4", "pqm-1", "mtl-2", "ets-4", "aat-6")
GOI_df <-
  GOI_annotate_heatmap(gene_names, time_resolved_rna_intestine_df$wikigene_name)
GOI_df
```


```{r}
time_resolved_rna_intestine_df %>% filter(wikigene_name %in% GOI_df$name)
```


```{r}
Boeck_intestine_RNA <-
  Heatmap(
    time_resolved_rna_intestine_matrix_scaled,
    cluster_columns = FALSE,
    show_row_names = FALSE,
    row_km = 5
  ) +
  rowAnnotation(foo = anno_mark(GOI_df$index, labels = GOI_df$name))
Boeck_intestine_RNA
# pdf(file = "./03_plots/200915_Boeck_RNA_Intestine.pdf", width = 7, height = 7)
# Boeck_intestine_RNA
# dev.off()
```

Filter heatmap for only transcription factors. This is very ugly, fix later.

```{r}
time_resolved_rna_intestine_matrix_scaled_TFONLY <-
  matrix_select(time_resolved_rna_intestine_matrix_scaled, wTF3.0$WBGeneID)

tf_GOI_df <-
  GOI_df %>% 
  left_join(time_resolved_rna_intestine_df, by = c("name" = "wikigene_name")) %>% 
  select(name:wbps_gene_id, -index) %>% filter(wbps_gene_id %in% wTF3.0$WBGeneID)
tf_GOI_df
tf_GOI_df <-
  GOI_annotate_heatmap(
    tf_GOI_df$wbps_gene_id,
    rownames(time_resolved_rna_intestine_matrix_scaled_TFONLY)
  ) %>% full_join(tf_GOI_df, by = c("name" = "wbps_gene_id"))

```


```{r}

Heatmap(
  time_resolved_rna_intestine_matrix_scaled_TFONLY,
  cluster_columns = FALSE,
  show_row_names = FALSE,
  row_split=6
) +
  rowAnnotation(foo = anno_mark(at = tf_GOI_df$index,
                                labels = tf_GOI_df$name.y))
```


# Import ELT-2 ChIP-seq binding data

```{r}
chip_df <-
  read_csv(file = "../01_ChIPseq_RNAseq_Integration/01_input/200719_annotatedPeaks.csv")
head(chip_df)
```

# Subset ELT-2 ChIP with literature Intestine Expression

Do this earlier in the code to have k4labels stored in the time_resolved_rna dataframe and subsequent subsetting

```{r}
chip_rna_df <- chip_df %>%
  select(name, cluster.description, WBGeneID)  %>%
  right_join(time_resolved_rna_intestine_df,
             by = c("WBGeneID" = "wbps_gene_id")) %>%
  replace_na(list("cluster.description" = "Not_Bound", "name" = "Not_Bound"))

chip_rna_df$cluster.description <-
  factor(
    chip_rna_df$cluster.description,
    levels = c(
      "Embryo_Specific",
      "Larval",
      "Increasing",
      "L3_High",
      "Not_Changing",
      "Not_Bound"
    )
  )
```

# Subset heatmap based on ELT-2 binding pattern

```{r chip_rna_matrix}
#### Handle duplicate rows created by 1:many gene:peak mapping

# match will return the first index of each non-redundant gene
nr_gene_name_ixs = match(unique(chip_rna_df$wikigene_name),chip_rna_df$wikigene_name)
#length(nr_gene_name_ixs)
#[1] 3286

chip_rna_df = chip_rna_df[nr_gene_name_ixs,]


chip_rna_matrix <-
  chip_rna_df %>% select(emb_510min:YA) %>% as.matrix()
  #chip_rna_df %>% select(emb_548min,emb_626min,L1,L2,L3,L4) %>% as.matrix()


#### Handle 0's and take the log
# 1. Just replace 0's as NAs so we can apply log(). Alternatively, we could do log(x + .01), but there are only ~300
chip_rna_matrix_na = chip_rna_matrix; 
chip_rna_matrix_na[0 == chip_rna_matrix_na] <- NA
# 2. Apply log()
chip_rna_matrix_log = log( chip_rna_matrix_na )
# 3. Do variances row-wise, make sure to set na.rm=T
rowvariances = apply(chip_rna_matrix_log, 1, var, na.rm=T)

range(rowvariances) # no NaNs
# 4. Plot distribution of row variances of the log data...
# You can do hist with 10,100,1000 breaks, and there will always be
# a dominant spike all the way on the lowest value. This is because the data elicit no natural lowest break, and can be demonstrated by ecdf, which is binless.
plot(ecdf(rowvariances)) # no "steps" anywhere, just a smooth curve
# therefore, we will choose to exclude the lowest 5% of the rows by their variance

chip_rna_matrix = chip_rna_matrix_log
rownames(chip_rna_matrix) <- chip_rna_df$wikigene_name
chip_rna_matrix_scaled <- row_scale(chip_rna_matrix) # calls base::scale() via RWC23_Functions.R

```

```{r print-gene-names-indices}
for (name in gene_names) {
  index <- which(rownames(chip_rna_matrix_scaled) == name)
  for (i in 1:length(index)) {
    print(c(name, index[i]))
  }
}
```
```{r heatmap_function}
BoeckRNA_ELT2_chip_Heatmap <- 
  function(subsetrows, 
           column_title,
           name = "Boeck Time Resolved RNA",                                
           row_split=chip_rna_df$cluster.description[subsetrows],
           clustering_distance_rows = "euclidean",
           clustering_method_rows = "complete",
           ...)
           
{
  
  ix=which(rownames(chip_rna_matrix_scaled)[subsetrows] %in% gene_names)
  chip_GOI_df = data.frame(name=rownames(chip_rna_matrix_scaled)[subsetrows][ix], index=ix)
  
  BoeckRNA_ELT2_chip <- Heatmap(
  chip_rna_matrix_scaled[subsetrows,],
  name=name,
  row_split = row_split,
  column_title = column_title,
  row_title = NULL,
  cluster_columns = FALSE,
  clustering_distance_rows = clustering_distance_rows,
  clustering_method_rows = clustering_method_rows,
  ...
) +
  rowAnnotation(
    ELT2_cluster = chip_rna_df$cluster.description[subsetrows],
    col = list(
      ELT2_cluster = c(
        "Embryo_Specific" = "#7570B3",
        "Larval" = "#1B9E77",
        "Increasing" = "#E7298A",
        "L3_High" = "#D95F02",
        "Not_Changing" = "#505050",
        "Not_Bound" = "yellow"
      )
    ),
    border = TRUE
  ) +  rowAnnotation(foo = anno_mark(at = chip_GOI_df$index,
                               labels = chip_GOI_df$name))
BoeckRNA_ELT2_chip
}
```


```{r BoeckRNA_ELT2_chip}
library(fpc) # for bootstrapping
library(vcd) # for mosaic
# 
# gene_names <-
#   c(
#     "flh-3",
#     "elt-7",
#     "clec-258",
#     "pqm-1",
#     "mtl-2",
#     "ets-4",
#     "aat-6",
#     "sre-23",
#     "Y45F10B.59",
#     "zmp-2",
#     "F08G5.3",
#     "Y113G7B.11",
#     "F40F9.10",
#     "coel-1",
#     "T25D10.4",
#     "M88.4",
#     "bcl-11",
#     "clec-44",
#     "lin-48",
#     "K01F9.2"
#   )

gene_names <- c("sre-23", "T25D10.4", "M88.4", "bcl-11", "aat-6", "pgp-1", "ftn-1")

real = apply(chip_rna_matrix_scaled, 1, function(x) { ! any(is.na(x)) }) # NA's introduced by log transform don't jive here
embryo_specific = chip_rna_df$cluster.description == "Embryo_Specific"
larval = chip_rna_df$cluster.description == "Larval"
l3_high = chip_rna_df$cluster.description == "L3_High"
increasing = chip_rna_df$cluster.description == "Increasing"

#changing = rowvariances > 0.1355294 # .05 thresh from chipseq

### ALL ###
qthreshold = quantile(rowvariances,.2)
changing = rowvariances > qthreshold
BoeckRNA_ELT2_chip_Heatmap(real & changing, sprintf("Log(RNA Timecourse), Variance > %.2f", qthreshold))

data.whole = chip_rna_matrix_scaled[real&changing,]
agglom='ward.D2'
hc = hclust(dist(data.whole), method =agglom )
# plot the hc height to get the screeplot
ggplot(NULL, aes(x=length(hc$height):1, y=hc$height)) +
    geom_point() + geom_line() +
    theme_bw() + labs(title="Scree Plot of hclust", x = "# of clusters", y="Height") + scale_x_continuous(limits=c(1,15)) 

nclust=3
# main 2 clusters are stable, 3rd unstable 
# set.seed(5)
# clusterboot( dist(data.whole), clustermethod=disthclustCBI,method=agglom,k=nclust)

subsetrows=real&changing
ix=which(rownames(chip_rna_matrix_scaled)[subsetrows] %in% gene_names)
  chip_GOI_df = data.frame(name=rownames(chip_rna_matrix_scaled)[subsetrows][ix], index=ix)

Heatmap(data.whole,
        column_title=sprintf("Log(RNA Timecourse), Variance > %.2f", qthreshold),
        name = "Boeck Time Resolved RNA",
        cluster_columns = F,
        row_order=hc$order,row_split=cutree(hc, nclust),show_row_names = F) +
  rowAnnotation(
    ELT2_cluster = chip_rna_df$cluster.description[real&changing],
    col = list(
      ELT2_cluster = c(
        "Embryo_Specific" = "#7570B3",
        "Larval" = "#1B9E77",
        "Increasing" = "#E7298A",
        "L3_High" = "#D95F02",
        "Not_Changing" = "#505050",
        "Not_Bound" = "yellow"
      )
    ),
    border = TRUE
  )+  rowAnnotation(foo = anno_mark(at = chip_GOI_df$index,
                               labels = chip_GOI_df$name))
hclusters = cutree(hc, nclust)
tab=table(hclusters, chip_rna_df$cluster.description[real&changing])
dimnames(tab) <- list("hclusters"=1:nclust, 
                      "ELT-2"=levels(chip_rna_df$cluster.description[real&changing]))
dimnames(tab)
mosaic(t(tab),shade=T,
       just_labels="right",
       rot_labels=c(0,0,0,0),
       offset_label=c(0,0,0,-.5),
       main  ="Time Resolved RNA associates with some \nELT-2 peak clusters")

```

```{r}
boeck_heatmap <- Heatmap(data.whole,
        column_title=sprintf("Log(RNA Timecourse), Variance > %.2f", qthreshold),
        name = "Boeck Time Resolved RNA",
        cluster_columns = F,
        row_order=hc$order,row_split=cutree(hc, nclust),show_row_names = F) + rowAnnotation(foo = anno_mark(at = chip_GOI_df$index,
                               labels = chip_GOI_df$name))


pdf(file = "03_plots/Boeck_Time_Resolved_RNA_Heatmap_geneAnnotated_201216.pdf", width = 7, height = 5)
boeck_heatmap
dev.off()
dev.set(dev.next())
```


```{r pearson-residuals}
corrplot(
  chisq.test(tab, simulate.p.value = TRUE)$stdres,
  is.corr = FALSE,
  cl.ratio = 0.35,
  cl.align = "r",
  tl.col = "black",
  tl.cex = 1,
  tl.srt = 45
)

```
```{r custom-mosaic}
boeck_pearson_residuals <-
  as.data.frame(chisq.test(tab, simulate.p.value = TRUE)$stdres)
boeck_pearson_residuals <-
  cbind(boeck_pearson_residuals,
        as.data.frame(chisq.test(tab, simulate.p.value = TRUE)$observed)$Freq)
colnames(boeck_pearson_residuals) <-
  c("hclusters", "ELT2_cluster", "Pearson_Residual", "Observed")
boeck_pearson_residuals$hclusters <-
  factor(
    boeck_pearson_residuals$hclusters,
    levels = c("3", "2", "1")
  )
ggplot(boeck_pearson_residuals,
       aes(x = ELT2_cluster, y = hclusters)) +
  geom_point(aes(size = Observed, color = Pearson_Residual)) + scale_color_gradient2(
    high = "red",
    low = "blue",
    mid = "white",
    midpoint = 0
  ) +
  theme_bw()
ggsave("03_plots/Boeck_ELT2clusters_moasiac_201216.pdf", width = 5, height = 3, useDingbats=FALSE)
```


```{r zoom-ins}
gene_names=rownames(chip_rna_matrix_scaled)[real & changing & embryo_specific]
(BoeckRNA_ELT2_chip_Heatmap(real & changing & embryo_specific, "Log(RNA Timecourse), No threshold on variance",row_split=2)->emb)
pdf(file="embryo.pdf",height=10.5,width=10.5)
print(emb)
dev.off()

gene_names=rownames(chip_rna_matrix_scaled)[real & changing & larval]
(BoeckRNA_ELT2_chip_Heatmap(real & changing & larval, "Log(RNA Timecourse), variance > 1 (top 20% data)",row_split=5)->larv)

pdf(file="larval.pdf",height=14,width=14)
print(larv)
dev.off()

gene_names=rownames(chip_rna_matrix_scaled)[real & changing & l3_high]
(BoeckRNA_ELT2_chip_Heatmap(real & changing & l3_high, "Log(RNA Timecourse), variance > 1 (top 20% data)",row_split=4)->L3plot)
pdf(file="L3.pdf",height=14,width=14)
print(L3plot)
dev.off()

gene_names=rownames(chip_rna_matrix_scaled)[real & changing & increasing]
(BoeckRNA_ELT2_chip_Heatmap(real & changing & increasing, "Log(RNA Timecourse), variance > 1 (top 20% data)",row_split=4)->inc)
pdf(file="increasing.pdf",height=21,width=21)
print(inc)
dev.off()

```

